### ISSUE TITLE
[CI] test failure in o.o.search.aggregations.bucket.composite.CompositeAggregatorTests.testWithKeywordLongAndMissingBucket

### ISSUE THREAD

**Issue by nknize** [opensearch-project/OpenSearch#1998]
Reproducible test failure related to #1942

```
REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.search.aggregations.bucket.composite.CompositeAggregatorTests.testWithKeywordLongAndMissingBucket" -Dtests.seed=7AEBC6A11333343F -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es-PY -Dtests.timezone=Pacific/Funafuti -Druntime.java=17
```

```
org.opensearch.search.aggregations.bucket.composite.CompositeAggregatorTests > testWithKeywordLongAndMissingBucket FAILED
    java.lang.AssertionError: expected:<3> but was:<4>
        at __randomizedtesting.SeedInfo.seed([7AEBC6A11333343F:9BD426C674ED5A70]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:647)
        at org.junit.Assert.assertEquals(Assert.java:633)
        at org.opensearch.search.aggregations.bucket.composite.CompositeAggregatorTests.lambda$testWithKeywordLongAndMissingBucket$93(CompositeAggregatorTests.java:1139)
        at org.opensearch.search.aggregations.bucket.composite.CompositeAggregatorTests.executeTestCase(CompositeAggregatorTests.java:2712)
        at org.opensearch.search.aggregations.bucket.composite.CompositeAggregatorTests.testSearchCase(CompositeAggregatorTests.java:2660)
        at org.opensearch.search.aggregations.bucket.composite.CompositeAggregatorTests.testWithKeywordLongAndMissingBucket(CompositeAggregatorTests.java:1128)
```

**Comment by nknize** [comment]
@penghuo can you have a look at this test failure?

**Comment by penghuo** [comment]
> @penghuo can you have a look at this test failure?

Ack

**Comment by penghuo** [comment]
The root cause of the issue is InternalBucket does not use MissingOrders. In bucket reduce phase, null bucket's order is less than others.
The test cases not always fail because the bug in bucket reduce phase which required at least 2 shards.
I am preparing PR now, will publish soon.

**Comment by penghuo** [comment]
@nknize Could we close this issue? Failed test already been fixed by #2005.
